libs/remote_storage/src/lib.rs

```diff
@@ -621,167 +636,41 @@ impl Debug for AzureConfig {
     }
 }
 
+fn deserialize_storage_class<'de, D: serde::Deserializer<'de>>(
+    deserializer: D,
+) -> Result<Option<StorageClass>, D::Error> {
+    Option::<String>::deserialize(deserializer).and_then(|s| {
+        if let Some(s) = s {
+            use serde::de::Error;
+            let storage_class = StorageClass::from_str(&s).expect("infallible");
+            #[allow(deprecated)]
+            if matches!(storage_class, StorageClass::Unknown(_)) {
+                return Err(D::Error::custom(format!(
+                    "Specified storage class unknown to SDK: '{s}'. Allowed values: {:?}",
+                    StorageClass::values()
+                )));
+            }
+            Ok(Some(storage_class))
+        } else {
+            Ok(None)
+        }
+    })
+}
+
 impl RemoteStorageConfig {
     pub const DEFAULT_TIMEOUT: Duration = std::time::Duration::from_secs(120);
 
     pub fn from_toml(toml: &toml_edit::Item) -> anyhow::Result<Option<RemoteStorageConfig>> {
-        let local_path = toml.get("local_path");
-        let bucket_name = toml.get("bucket_name");
-        let bucket_region = toml.get("bucket_region");
-        let container_name = toml.get("container_name");
-        let container_region = toml.get("container_region");
-
-        let use_azure = container_name.is_some() && container_region.is_some();
-
-        let default_concurrency_limit = if use_azure {
-            DEFAULT_REMOTE_STORAGE_AZURE_CONCURRENCY_LIMIT
-        } else {
-            DEFAULT_REMOTE_STORAGE_S3_CONCURRENCY_LIMIT
+        let toml_edit::Item::Table(toml) = toml else {
+            bail!("toml not a table");
         };
-        let concurrency_limit = NonZeroUsize::new(
-            parse_optional_integer("concurrency_limit", toml)?.unwrap_or(default_concurrency_limit),
-        )
-        .context("Failed to parse 'concurrency_limit' as a positive integer")?;
-
-        let max_keys_per_list_response =
-            parse_optional_integer::<i32, _>("max_keys_per_list_response", toml)
-                .context("Failed to parse 'max_keys_per_list_response' as a positive integer")?
-                .or(DEFAULT_MAX_KEYS_PER_LIST_RESPONSE);
```

Comment:
This logic is lost by your changes. The tests you added use the `DEFAULT_MAX_KEYS_PER_LIST_RESPONSE` but they only work because `DEFAULT_MAX_KEYS_PER_LIST_RESPONSE = Default::default() = None`.

Reply:
=> probably best to review the various `DEFAULT_` statics & consts in this file and clean them up, can you do that as part of this PR? I think you have the best grip on remote storage config ATM.

Reply:
> This logic is lost by your changes. 

restored, with the fn method as serde has no default literals (yet).

> clean them up, can you do that as part of this PR?

What do you mean by that? It might make sense to move them into associated consts though.

Reply:
> What do you mean by that? It might make sense to move them into associated consts though.

Yeah, something like associated consts or just a `serde(default = ...)` .

Not a blocker, feel free to dismiss